src/esp/physics/bullet/BulletRigidObject.cpp

```diff
@@ -177,6 +177,8 @@ bool BulletRigidObject::initializeObject(
   if (joinCollisionMeshes) {
     btTransform t;
     t.setIdentity();
+    bObjectConvexShapes_.back()->setMargin(0.0);
```

Comment:
I haven't dug too deep in the physics code yet but why is this here and not in constructBulletCompoundFromMeshes

Reply:
`constructBulletCompoundFromMeshes` is a recursive function. 

With the `join` flag, as in this case, it is building up a single convex shape from all components. This will not be complete until after the recursion is done. 

Without the `join` flag, each level of recursion builds a new object, so it makes sense to set margin upon construction in that case.

Reply:
The constructBulletCompoundFromMeshes code might be easier to read with two functions. A top-level function and a recursive helper. It feels like functionality that should be part of constructBulletCompoundFromMeshes is part of initializeObject.

Reply:
Fair point. 👍 
A refactor like this should probably be a separate PR after this bug fix is merged.